bundles/com.amazonaws.eclipse.cloudformation/src/com/amazonaws/eclipse/explorer/cloudformation/wizard/CreateStackWizardFirstPage1.java

```diff
@@ -0,0 +1,792 @@
+        addStatusDecorator(templateURLText, new ChainValidator<String>(templateValidated, useTemplateUrl,
+                templateValidator));
+
+        // Finally provide aggregate status reporting for the entire wizard page
+        final AggregateValidationStatus aggregateValidationStatus = new AggregateValidationStatus(bindingContext,
+                AggregateValidationStatus.MAX_SEVERITY);
+
+        aggregateValidationStatus.addChangeListener(new IChangeListener() {
+
+            @Override
+            public void handleChange(ChangeEvent event) {
+                Object value = aggregateValidationStatus.getValue();
+                if ( value instanceof IStatus == false )
+                    return;
+
+                IStatus status = (IStatus) value;
+                if ( status.isOK() ) {
+                    setErrorMessage(null);
+                    if (wizard.getDataModel().getMode() == Mode.EstimateCost) {
+                        setMessage(ESTIMATE_COST_OK_MESSAGE, Status.OK);
+                    } else {
+                        setMessage(OK_MESSAGE, Status.OK);
+                    }
+                } else if (status.getSeverity() == Status.WARNING) {
+                    setErrorMessage(null);
+                    setMessage(status.getMessage(), Status.WARNING);
+                } else if (status.getSeverity() == Status.ERROR) {
+                    setErrorMessage(status.getMessage());
+                }
+
+                setComplete(status.isOK());
+            }
+        });
+    }
+
+    /**
+     * Adds a control status decorator for the control given.
+     */
+    private void addStatusDecorator(final Control control, ValidationStatusProvider validationStatusProvider) {
+        ControlDecoration decoration = new ControlDecoration(control, SWT.TOP | SWT.LEFT);
+        decoration.setDescriptionText("Invalid value");
+        FieldDecoration fieldDecoration = FieldDecorationRegistry.getDefault().getFieldDecoration(
+                FieldDecorationRegistry.DEC_ERROR);
+        decoration.setImage(fieldDecoration.getImage());
+        new DecorationChangeListener(decoration, validationStatusProvider.getValidationStatus());
+    }
+
+    @Override
+    public boolean isPageComplete() {
+        return complete;
+    }
+
+    private void setComplete(boolean complete) {
+        this.complete = complete;
+        if ( getWizard().getContainer() != null && getWizard().getContainer().getCurrentPage() != null )
+            getWizard().getContainer().updateButtons();
+    }
+
+    /**
+     * Loads all SNS topics into the dropdown given
+     */
+    private void loadTopics(final Combo snsTopicCombo) {
+        snsTopicCombo.setItems(new String[] { "Loading..." });
+        new Thread() {
+
+            @Override
+            public void run() {
+                AmazonSNS sns = AwsToolkitCore.getClientFactory().getSNSClient();
+                ListTopicsResult topicsResult = sns.listTopics();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.